OpenSim/Sandbox/MatlabScripts/c3d_reading.m

```diff
@@ -1,124 +1,214 @@
+        % rotate marker data
+        markers = rotateTableData(markers, rotationMatrix);
+        % rotate force data
+        forces_rotated = rotateTableData(forces,rotationMatrix);
+    end
+end
+%% Print the rotated markers to trc file
+% make trc adapter and write marker tables to file.
+% TRCFileAdapter requires the table to have DataRate and Units meta data.
+% In this case, we made a copy of markers table, so the meta data got
+% copied. If you make a new table, you will need to set these meta data
+% keys before using TRCFileAdapter.
+% ie markers.addTableMetaDataString('DataRate', '250')
+TRCFileAdapter().write(markers,[filename '.trc']);
+
+%% Print the force data as a Vec3 sto file and a flattened doubles sto file
+% make postfix string vector for naming colomns
+postfix = StdVectorString();
+postfix.add('_x');
+postfix.add('_y');
+postfix.add('_z');
+
+%% flatten the Vec3 table to a doubles table. 
+% This converts a table of Vec3's into a flat table of doubles. Default
+% column names have '_1', '_2', '_3' added. Here we specify the postfic as
+% '_x', '_y', and 'z'.
+forces_flattened = forces.flatten(postfix);
+% make a sto adapter and write the forces table to file.
+STOFileAdapter().write(forces_flattened,[filename '.mot']);
 
+end
 
-TimeSeriesTableVec3::DependentsMetaData marker_dep_metadata{};
-marker_dep_metadata.setValueArrayForKey("labels", marker_labels);
-marker_table.setDependentsMetaData(marker_dep_metadata);
 
-double time_step{1.0 / acquisition->GetPointFrequency()};
-for f = 1:0 
-    
-    
-    rowVector = RowVector()
-    
-    
-    
-    for(auto it = marker_pts->Begin();
-        it != marker_pts->End();
-        ++it) {
-        auto pt = *it;
-        row[m++] = SimTK::Vec3{pt->GetValues().coeff(f, 0),
-                               pt->GetValues().coeff(f, 1),
-                               pt->GetValues().coeff(f, 2)};
-    }
+function table_rotated = rotateTableData(table, rotationMatrix)
 
-    
-marker_table.appendRow(0 + f * time_step, row);
+import org.opensim.modeling.*
 
+nLabels = table.getNumColumns();
+nRows = table.getNumRows();
 
-for iMarker = 0 : length(markerdata) - 1
+table_rotated = table();
+
+for it = 0 : nLabels - 1
 
     % get the column data for the marker
-    marker = markers.updDependentColumn(markerlabels(iMarker+1) );
+    table_column = table_rotated.updDependentColumnAtIndex( it   );
 
     % go through each element of the table column, rotate the Vec3, and write
     % back to the column.
-    for iRow = 0 : markers.getNumRows - 1
+    for iRow = 0 : nRows - 1
         % get Matlab vector marker position
-        vectorData = [marker.getElt(0,iRow).get(0)...
-                      marker.getElt(0,iRow).get(1)...
-                      marker.getElt(0,iRow).get(2)];
+        vectorData = [table_column.getElt(0,iRow).get(0)...
+                      table_column.getElt(0,iRow).get(1)...
+                      table_column.getElt(0,iRow).get(2)];
```

Comment:
Matlab can't perform some of the operations here (`auto rotatedV = r*packedV`). Any ideas on how I can rotate a Vector of Vec3's?